src/components/PlaylistsPage.vue

```diff
@@ -94,22 +106,96 @@ export default {
-                        Authorization: this.getAuthToken(),
-                        "Content-Type": "application/json",
-                    },
-                }).then(json => {
-                    if (json.error) alert(json.error);
-                    else this.fetchPlaylists();
+            if (!name) return;
+            this.createPlaylist(name).then(json => {
+                if (json.error) alert(json.error);
+                else this.fetchPlaylists();
+            });
+        },
+        async createPlaylist(name) {
+            let json = await this.fetchJson(this.authApiUrl() + "/user/playlists/create", null, {
+                method: "POST",
+                body: JSON.stringify({
+                    name: name,
+                }),
+                headers: {
+                    Authorization: this.getAuthToken(),
+                    "Content-Type": "application/json",
+                },
+            });
+            return json;
+        },
+        exportPlaylists() {
+            if (!this.playlists) return;
+            let json = {
+                format: "Piped",
+                version: 1,
+                playlists: [],
+            };
+            let playlistsSize = this.playlists.length;
+            for (var i = 0; i < playlistsSize; i++) {
+                this.fetchPlaylistJson(this.playlists[i].id, playlist => {
+                    json.playlists.push(playlist);
+                    if (playlistsSize != json.playlists.length) return;
+                    this.download(JSON.stringify(json), "playlists.json", "application/json");
                 });
+            }
+        },
+        async fetchPlaylistJson(playlistId, onSuccess) {
+            let playlist = await this.fetchJson(this.authApiUrl() + "/playlists/" + playlistId);
+            let playlistJson = {
+                name: playlist.name,
+                // possible other types: history, watch later, ...
+                type: "playlist",
+                // as Invidious supports public and private playlists
+                visibility: "private",
+                // list of the videos, starting with "https://youtube.com" to clarify that those are YT videos
+                videos: [],
+            };
+            for (var i = 0; i < playlist.relatedStreams.length; i++) {
+                playlistJson.videos.push("https://youtube.com" + playlist.relatedStreams[i].url);
+            }
+            onSuccess(playlistJson);
+        },
+        importPlaylists() {
+            const file = this.$refs.fileSelector.files[0];
+            file.text().then(text => {
+                let playlists = JSON.parse(text).playlists;
+                if (!playlists.length) {
+                    alert(this.$t("actions.no_valid_playlists"));
+                    return;
+                }
+                let importedCount = 0;
+                for (var i = 0; i < playlists.length; i++) {
+                    this.createPlaylistWithVideos(playlists[i], () => {
```

Comment:
Instead of doing this, you should add all the promises to an array, and then call `Promise.all()`